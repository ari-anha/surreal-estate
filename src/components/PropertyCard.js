import React from "react";
import PropTypes from "prop-types";
import "../styles/PropertyCard.css";

import { GiFamilyHouse } from "react-icons/gi";
import { FaBed, FaBath } from "react-icons/fa";
import { FiMail } from "react-icons/fi";

const PropertyCard = ({
  title,
  type,
  bathrooms,
  bedrooms,
  price,
  city,
  email,
}) => {
  return (
    <div className="property-card">
      <h2 className="icon">
        <GiFamilyHouse />
      </h2>
      <h3 className="title" data-testid="title-id">
        {title}
      </h3>

      <h4 className="type-city" data-testid="type-city-id">
        {type} - {city}
      </h4>
      <h4 className="bathrooms" data-testid="bathrooms-id">
        <FaBath /> Bathrooms: {bathrooms}
      </h4>
      <h4 className="bedrooms" data-testid="bedrooms-id">
        <FaBed /> Bedrooms: {bedrooms}
      </h4>
      <h4 className="price" data-testid="price-id">
        Â£{price}
      </h4>
      <button className="emailbutton" data-testid="emailbutton-id">
        <a href={`mailto:${email}`}>
          <FiMail />
          Send email
        </a>
      </button>
    </div>
  );
};

export default PropertyCard;
