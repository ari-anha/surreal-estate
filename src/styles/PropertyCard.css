.property-card {
  width: 100%;
}
