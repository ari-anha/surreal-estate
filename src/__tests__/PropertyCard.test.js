import "@testing-library/jest-dom/extend-expect";
import React from "react";
import { render } from "@testing-library/react";
import PropertyCard from "../components/PropertyCard";

describe("PropertyCard", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <PropertyCard
        title="mockTitle"
        type="Bungalow"
        bathrooms={9}
        bedrooms={6}
        price={200000}
        city="Leeds"
        email="mockemail@email.com"
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct props", () => {
    const { getByTestId } = render(
      <PropertyCard
        title="mockTitle"
        type="Bungalow"
        bathrooms={9}
        bedrooms={6}
        price={200000}
        city="Leeds"
        email="mockemail@email.com"
      />
    );

    expect(getByTestId("title-id")).toHaveClass("title");
    expect(getByTestId("type-city-id")).toHaveClass("type-city");
    expect(getByTestId("bathrooms-id")).toHaveClass("bathrooms");
    expect(getByTestId("price-id")).toHaveClass("price");
    expect(getByTestId("emailbutton-id")).toHaveClass("emailbutton");
  });
});
