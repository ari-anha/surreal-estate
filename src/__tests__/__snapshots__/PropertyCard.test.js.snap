// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <h2
      class="icon"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M55.379 25l-28.4 142H172.27L256 83.271 339.729 167H485.02l-28.4-142zM256 108.727L179.729 185H41v302h158v-87c0-18.25 7.166-33.077 18.021-42.727C227.877 347.624 242 343 256 343s28.123 4.624 38.979 14.273C305.834 366.923 313 381.75 313 400v87h158V185H332.271zm0 38.544l57 57V297H199v-92.729zm0 25.456l-39 39V279h78v-67.271zM71 199h98v98H71zm272 0h98v98h-98zM89 217v30h62v-30zm272 0v30h62v-30zM89 265v14h62v-14zm272 0v14h62v-14zM71 359h98v98H71v-98zm272 0h98v98h-98v-98zm-87 2c-10 0-19.877 3.376-27.021 9.727C221.834 377.077 217 386.25 217 400v87h78v-87c0-13.75-4.834-22.923-11.979-29.273C275.877 364.376 266 361 256 361zM89 377v62h62v-62zm272 0v62h62v-62z"
        />
      </svg>
    </h2>
    <h3
      class="title"
      data-testid="title-id"
    >
      mockTitle
    </h3>
    <h4
      class="type-city"
      data-testid="type-city-id"
    >
      Bungalow - Leeds
    </h4>
    <h4
      class="bathrooms"
      data-testid="bathrooms-id"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M32,384a95.4,95.4,0,0,0,32,71.09V496a16,16,0,0,0,16,16h32a16,16,0,0,0,16-16V480H384v16a16,16,0,0,0,16,16h32a16,16,0,0,0,16-16V455.09A95.4,95.4,0,0,0,480,384V336H32ZM496,256H80V69.25a21.26,21.26,0,0,1,36.28-15l19.27,19.26c-13.13,29.88-7.61,59.11,8.62,79.73l-.17.17A16,16,0,0,0,144,176l11.31,11.31a16,16,0,0,0,22.63,0L283.31,81.94a16,16,0,0,0,0-22.63L272,48a16,16,0,0,0-22.62,0l-.17.17c-20.62-16.23-49.83-21.75-79.73-8.62L150.22,20.28A69.25,69.25,0,0,0,32,69.25V256H16A16,16,0,0,0,0,272v16a16,16,0,0,0,16,16H496a16,16,0,0,0,16-16V272A16,16,0,0,0,496,256Z"
        />
      </svg>
       Bathrooms: 9
    </h4>
    <h4
      class="bedrooms"
      data-testid="bedrooms-id"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 640 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M176 256c44.11 0 80-35.89 80-80s-35.89-80-80-80-80 35.89-80 80 35.89 80 80 80zm352-128H304c-8.84 0-16 7.16-16 16v144H64V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v352c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16v-48h512v48c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16V240c0-61.86-50.14-112-112-112z"
        />
      </svg>
       Bedrooms: 6
    </h4>
    <h4
      class="price"
      data-testid="price-id"
    >
      Â£200000
    </h4>
    <button
      class="emailbutton"
      data-testid="emailbutton-id"
    >
      <a
        href="mailto:mockemail@email.com"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z"
          />
          <polyline
            points="22,6 12,13 2,6"
          />
        </svg>
        Send email
      </a>
    </button>
  </div>
</DocumentFragment>
`;
